app/main/views/direct_award.py

```diff
@@ -188,3 +190,52 @@ def get_project_search(project_id, search_id):
 @main.route('/direct-award/projects/<int:project_id>/services', methods=['GET'])
 def list_project_services(project_id):
     raise NotImplementedError()
+
+
+@main.route('/direct-award/projects/<int:project_id>/lock', methods=['POST'])
+def lock_project(project_id):
+    updater_json = validate_and_return_updater_request()
+
+    project = DirectAwardProject.query.filter(
+        DirectAwardProject.id == project_id
+    ).first_or_404()
+
+    if project.locked_at:
+        abort(400, 'Project has already been locked: {}'.format(project_id))
+
+    search = DirectAwardSearch.query.filter(
+        DirectAwardSearch.project_id == project_id
+    ).first_or_404()
+
+    now = datetime.datetime.utcnow()
+    service_ids = list(map(lambda x: x['id'], search_api_client.search_services_from_url_iter(search.search_url)))
```

Comment:
I'd have preferred `= [x['id'] for x in search_api_client.search_services_from_url_iter(search.search_url)]` .. I think?

Reply:
:+1: